extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsStorageDruidModule.java

```diff
@@ -126,5 +129,7 @@ public void configure(Binder binder)
     Binders.taskLogsBinder(binder).addBinding("hdfs").to(HdfsTaskLogs.class);
     JsonConfigProvider.bind(binder, "druid.indexer.logs", HdfsTaskLogsConfig.class);
     binder.bind(HdfsTaskLogs.class).in(LazySingleton.class);
+    LifecycleModule.register(binder, HdfsStorageAuthentication.class);
```

Comment:
@drcrallen thanks, i have added `binder.bind(HdfsStorageAuthentication.class).in(ManageLifecycle.class);` but still the life cycle starts before loading the extensions. any ideas ?